### DISCUSSION TITLE
Can't make xmlrpc-c work on Mac

### DISCUSSION THREAD

**Discussion by glitchtracker** [Homebrew/discussions#1645] [Everyday usage]
Hi,
I installed the xmlrpc-c  homebrew formula, but it doesn't seem to work.
The command line don't understand the command xmlrpc
Homebrew says it installed it correctly
How to make it work, I tried on macOS 10.12 Sierra and Big Sur, not "working"

I mean that the command xmlrpc doesn't install, the doc says

"Because the utility programs are not essential to the package, the default install tools don't install them at all. The builder does build them, though, and if you build Xmlrpc-c from source, you will find them all in the tools/ directory."

Unfortunately, I don't know to build it on mac, hence the fact that I'm relying on homebrew, is thre a trick to mac the "tools" install on mac ?

Thanks

**Comment by cho-m** [comment]
Currently, Homebrew's `xmlrpc-c` formula does not build the utility tools.
This is the default `make` behavior as commented in the `GNUmakefile`:
```
# The reason we don't build tools and examples by default is that they
# contain executables, which require significantly more from the
# environment to build than libraries.  Ergo, they are signficantly
# more likely to fail to build.  Indeed, when 'tools' was built by
# default, the majority of the reported build problems were with that.
# Since they are ancillary to the package, building them by default is
# not worth causing the whole build to fail.
```

In order to actually build the tools, you would need to use either `make all` or `cd tools && make` and then the xmlrpc tool would be built inside the local `tools/xmlrpc/xmlrpc` path.

**I am not familiar with the common usage of `xmlrpc-c`, so I cannot comment much on whether Homebrew's formula should do this or not.**

---

The following are examples of modifications to Formula to build/install all the tools (`xml-rpc-api2cpp`, `xml-rpc-api2txt`, `xmlrpc`, `xmlrpc_cpp_proxy`, `xmlrpc_dumpserver`, `xmlrpc_parsecall`, `xmlrpc_pstream`, `xmlrpc_transport`):

Example 1:
```diff
@@ -22,8 +22,9 @@ class XmlrpcC < Formula
                           "--prefix=#{prefix}"

     # xmlrpc-config.h cannot be found if only calling make install
-    system "make"
+    system "make", "all"
     system "make", "install"
+    system "make", "-C", "tools", "install"
   end
```

Example 2:
```diff
@@ -24,6 +24,10 @@ class XmlrpcC < Formula
     # xmlrpc-config.h cannot be found if only calling make install
     system "make"
     system "make", "install"
+    cd "tools" do
+      system "make"
+      system "make", "install"
+    end
   end
```

---

If the tools aren't common enough to be accepted to Homebrew's repo, then you can always copy/modify a Formula into your personal Tap.

**Reply by glitchtracker** [reply]
Thanks, so now I need to understand how to create a tap and use it…
Or maybe you could create one, I think it will help people

**Reply by carlocab** [reply]
Here are some docs on taps to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Comment by glitchtracker** [comment]
For now I just copied the original formula but I have this error 
 
Installing xmlrpc-c-all from glitchtracker/xmlrpc-c-all
==> Downloading https://ghcr.io/v2/homebrew/core/xmlrpc-c-all/manifests/1.51.07
#=#=#                                                                         
curl: (22) The requested URL returned error: 404 
Error: Failed to download resource "xmlrpc-c-all_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/xmlrpc-c-all/manifests/1.51.07

**Reply by nandahkrishna** [reply]
You'll have to remove the bottle block.

**Reply by nandahkrishna** [reply]
Using `brew extract` will help avoid this, it removes the bottle block and copies over the formula to your tap. All you have to do then is change its name (filename and class name).
